### PR TITLE
Some perfomance improvement

### PR DESCRIPTION
**Pull request by jonilviv**
- Added .NET platform compiler target: for better performance it is good, when all referenced libraries are compiled in same .NET target framework (no runtime switches).
- Added build optimizations for RELEASE build. As developer I want to have optimized code in release and I don't want to have PDB files in release build.

### CODE REVIEW

**Comment by jonilviv** [comment]
@Mpdreamz could you please review?

File: src/ShellProgressBar/ShellProgressBar.csproj [hunk 1]
```diff
@@ -23,4 +23,18 @@
...
+    <DebugType>full</DebugType>
+    <DebugSymbols>true</DebugSymbols>
+    <Optimize>false</Optimize>
+    <EnvironmentName>Development</EnvironmentName>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <DebugType>none</DebugType>
+    <DebugSymbols>false</DebugSymbols>
+    <Optimize>true</Optimize>
+    <EnvironmentName>Production</EnvironmentName>
```
Comment (Mpdreamz) [inline src/ShellProgressBar/ShellProgressBar.csproj:37]: `EnvironmentName` is aspnetcore specific and has no meaning here as far as i know.

File: src/ShellProgressBar/ShellProgressBar.csproj [hunk 2]
```diff
@@ -23,4 +23,18 @@
...
+  <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
+    <DebugType>full</DebugType>
+    <DebugSymbols>true</DebugSymbols>
+    <Optimize>false</Optimize>
+    <EnvironmentName>Development</EnvironmentName>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <DebugType>none</DebugType>
+    <DebugSymbols>false</DebugSymbols>
+    <Optimize>true</Optimize>
```
Comment (Mpdreamz) [inline src/ShellProgressBar/ShellProgressBar.csproj:36]: This is already be the default when building in `Release` mode.

File: src/ShellProgressBar/ShellProgressBar.csproj [hunk 3]
```diff
@@ -23,4 +23,18 @@
...
-</Project>
+
+  <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
+    <DebugType>full</DebugType>
+    <DebugSymbols>true</DebugSymbols>
+    <Optimize>false</Optimize>
+    <EnvironmentName>Development</EnvironmentName>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <DebugType>none</DebugType>
```
Comment (Mpdreamz) [inline src/ShellProgressBar/ShellProgressBar.csproj:34]: The default for .NET core is portable PDB files to be included, These are lighter and provide a much better experience out the box IMO

**Comment by Mpdreamz** [comment]
Most of these are already default, and I think building (portable) PDB files even in release mode and shipping them wth nuget package is a better experience OOTB for everyone.

Thanks for the PR!

**Pull request closed without merging**